src/Functions/parseTimeDelta.cpp

```diff
@@ -0,0 +1,243 @@
+
+        bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+        size_t getNumberOfArguments() const override { return 0; }
+
+        DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+        {
+            if (arguments.empty())
+                throw Exception(
+                    "Number of arguments for function " + getName() + " doesn't match: passed " + toString(arguments.size())
+                        + ", should be 1.",
+                    ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+            if (arguments.size() > 1)
+                throw Exception(
+                    "Number of arguments for function " + getName() + " doesn't match: passed " + toString(arguments.size())
+                        + ", should be at most 1.",
+                    ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+            const IDataType & type = *arguments[0];
+
+            if (!isString(type))
+                throw Exception("Cannot format " + type.getName() + " as time string", ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+            return std::make_shared<DataTypeFloat64>();
+        }
+
+        bool useDefaultImplementationForConstants() const override { return true; }
+
+        ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+        {
+            auto col_to = ColumnFloat64::create();
+            auto & res_data = col_to->getData();
+
+            for (size_t i = 0; i < input_rows_count; ++i)
+            {
+                std::string_view str{arguments[0].column->getDataAt(i)};
+                Int64 length = str.length();
+                Int64 fast = 0;
+                Int64 slow = 0;
+                Int64 last_pos = 0;
+                Float64 result = 0;
+
+                /// ignore '.' and ' ' at the end of string
+                for (Int64 j = length - 1; j >= 0; j--)
+                {
+                    if (str[j] != '.' && str[j] != ' ')
+                    {
+                        last_pos = j;
+                        break;
+                    }
+                }
+
+                /// no valid characters
+                if (last_pos == 0)
+                {
+                    res_data.emplace_back(0);
+                    continue;
+                }
+
+                while (fast <= last_pos)
+                {
+                    ScanIgnoreCharacters(str, fast, last_pos);
+                    slow = fast;
+
+                    /// scan unsigned integer
+                    if (!ScanUnsignedInteger(str, fast, last_pos))
+                    {
+                        throw Exception(
+                            "bad auguments invalid expression for function " + getName() + " find number failed: " + String(str) + ".",
+                            ErrorCodes::BAD_ARGUMENTS);
```

Comment:
Minor comment: it would be better to use formatting instead of concatinating strings. Also let's make exception message better - start message with uppercase character and don't write `bad arguments` because we use `BAD_ARGUMENTS` error code (here and in other exceptions):
```suggestion
                        throw Exception(
                               ErrorCodes::BAD_ARGUMENTS,
                               "Invalid expression for function {}, failed to parse number here: {}", getName(), String(str));
```